Add preference combo to change severity of 'missing project encoding'

ingomohr opened the pull request:
This introduces a new combo-box entry to the "Workspace" preference page
to modify the severity used for the check for "missing project-specific
encoding".
See issue #198.

ingomohr commented:
Pls note I created this PR to make integration testing of related [Backend PR 167](https://github.com/eclipse-platform/eclipse.platform.resources/pull/167) easier.

This PR is blocked by that PR in the platform repo.

trancexpress commented on bundles/org.eclipse.ui.ide/src/org/eclipse/ui/internal/ide/IDEWorkbenchMessages.java:
```diff
@@ -9,23 +9,24 @@
  * SPDX-License-Identifier: EPL-2.0
```
Would be nice to not do unnecessary formatting changes, at least not in combination with actual code changes that matter.

If this is done by the formatter, you can "undo" it with Ctrl+Z (or whichever undo hotkey you use) after a save.

ingomohr replied:
Thank you for the feedback. I wasn't sure about this change and whether everything on a PR is still supposed to go into one single commit. I considered the formatting of the contributors list broken and aimed to fix it so that it would match the other classes I'd seen.

I reverted that change now, so that the only change in the list is the new entry in the list.

trancexpress commented on bundles/org.eclipse.ui.ide/src/org/eclipse/ui/internal/ide/dialogs/IDEWorkspacePreferencePage.java:
```diff
@@ -197,6 +200,25 @@ private void createMissingNaturePref(Composite parent) {
 	}
 
+	/**
+	 * @param comboParent
```
Please remove comment.

ingomohr replied:
removed that jdoc with (even wrong) parameter name. Thanks for finding.

(I had added this for consistency with the other methods - but I agree that the comment doesn't add any value. Happy to drop it)

trancexpress commented on bundles/org.eclipse.ui.ide/src/org/eclipse/ui/internal/ide/messages.properties:
```diff
@@ -550,6 +550,8 @@ IDEWorkspacePreference_workspaceName=Wor&kspace name:
 IDEWorkbenchPreference_workbenchSystemExplorer=Command for launching system e&xplorer:
 IDEWorkspacePreference_UnknownNatureSeverity=Report unknown project n&ature as:
 IDEWorkspacePreference_UnknownNatureSeverity_Ignore=Ignore
+IDEWorkspacePreference_UnknownEncodingSeverity=Report missing project &encoding as:
```
1. "Show workspace nam&e"
2. "New text line delimiter" -> "D&efault"

Options uses &e in the same dialog, I'm not sure how this collision is meant to be resolved (Alt+E results in the "Show workspace name" checkbox being interacted with). @iloveeclipse ? If there is already 1 collision that cannot be worked around, no point in adding another one.

ingomohr replied:
True.

I admittedly didn't test the mnemonic at all (apparently, there is [no mnemonic support in SWT for macOS](https://bugs.eclipse.org/bugs/show_bug.cgi?id=399554)).

Do we want to choose some other char? If so: which one is best? Or is it feasible to not specify one at all?

ingomohr replied:
What about the c?

iloveeclipse commented:
/rebase

iloveeclipse commented:
> /rebase

Ignore that, I've forgot the other repo change will be visible on next day only. 
But in general, it is nicer (less work for reviewers) if PR's are rebased on latest master.

ingomohr committed:
Add preference combo to change severity of 'missing project encoding'

This introduces a new combo-box entry to the "Workspace" preference page
to modify the severity used for the check for "missing project-specific
encoding".
See issue #166 in repo eclipse.platform.resources

Pls note that this accesses internal code as suggested in the review
comment https://github.com/eclipse-platform/eclipse.platform.resources/pull/167/files/aaec47ecd62735a41b5149580bf87735f44f2093#r927372850

ingomohr commented:
> > /rebase
> 
> Ignore that, I've forgot the other repo change will be visible on next day only. But in general, it is nicer (less work for reviewers) if PR's are rebased on latest master.

Thanks. I didn't know about the "next day" thing. I still rebased my PR (before I saw your feedback here b/c my change was already 14 days old on my local branch).

iloveeclipse commented on bundles/org.eclipse.ui.ide/src/org/eclipse/ui/internal/ide/messages.properties:
```diff
@@ -550,6 +551,8 @@ IDEWorkspacePreference_workspaceName=Wor&kspace name:
 IDEWorkbenchPreference_workbenchSystemExplorer=Command for launching system e&xplorer:
 IDEWorkspacePreference_UnknownNatureSeverity=Report unknown project n&ature as:
 IDEWorkspacePreference_UnknownNatureSeverity_Ignore=Ignore
+IDEWorkspacePreference_UnknownEncodingSeverity=Report missing project en&coding as:
```
```suggestion
IDEWorkspacePreference_UnknownEncodingSeverity=Rep&ort missing project encoding as:
```

iloveeclipse committed:
Fixed mnemonic
